src/ring_theory/witt_vector/isocrystal.lean

```diff
@@ -0,0 +1,204 @@
+/-
+Copyright (c) 2022 Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Heather Macbeth
+-/
+
+import ring_theory.witt_vector.frobenius_fraction_field
+
+/-!
+
+## F-isocrystals over a perfect field
+
+When `k` is an integral domain, so is `𝕎 k`, and we can consider its field of fractions `K(p, k)`.
+The endomorphism `witt_vector.frobenius` lifts to `φ : K(p, k) → K(p, k)`; if `k` is perfect, `φ` is
+an automorphism.
+
+Let `k` be a perfect integral domain. Let `V` be a vector space over `K(p,k)`.
+An *isocrystal* is a bijective map `V → V` that is `φ`-semilinear.
+A theorem of Dieudonné and Manin classifies the finite-dimensional isocrystals over algebraically
+closed fields. In the one-dimensional case, this classification states that the isocrystal
+structures are parametrized by their "slope" `m : ℤ`.
```

Comment:
Can you add a description of the standard slope `m` isocrystal? Just saying what the Frobenius is.

Reply:
I've pushed another sentence. "Any one-dimensional isocrystal is isomorphic to `φ(p^m • x) : K(p,k) → K(p,k)` for some `m`." I'm blurring the line between "an isocrystal is a bijective map" as written in the header and "an isocrystal is a vector space equipped with a map" as defined in the code, do you think this is okay? (@hrmacbeth ?)

Reply:
Yes, thanks!